deltachat-ios/Chat/Views/MessageLabel.swift

```diff
@@ -452,49 +452,65 @@ open class MessageLabel: UILabel {
 
     }
 
-  open func handleGesture(_ touchLocation: CGPoint) -> Bool {
-
-        guard let index = stringIndex(at: touchLocation) else { return false }
+    internal func detectLink(_ touchLocation: CGPoint) -> (DetectorType, NewMessageTextCheckingType)? {
+        guard let index = stringIndex(at: touchLocation) else { return nil }
 
         for (detectorType, ranges) in rangesForDetectors {
             for (range, value) in ranges {
                 if range.contains(index) {
-                    handleGesture(for: detectorType, value: value)
-                    return true
+                    return (detectorType, value)
                 }
             }
         }
-        return false
+
+        return nil
     }
 
-    /// swiftlint:disable cyclomatic_complexity
-    private func handleGesture(for detectorType: DetectorType, value: NewMessageTextCheckingType) {
+    internal func getCopyableLinkText(_ touchLocation: CGPoint) -> String? {
+        guard let (_, value) = detectLink(touchLocation) else { return nil }
+
+        switch value {
+        case let .link(url):
+            guard let url = url else { return nil }
+            return url.absoluteString
+        case let .phoneNumber(phoneNumber):
+            return phoneNumber
+        case let .custom(_, match):
+            return match
+        case .addressComponents, .date, .transitInfoComponents:
+            return nil
+        }
+    }
+
+    internal func handleGesture(_ touchLocation: CGPoint) -> Bool {
```

Comment:
What's the idea behind the boolean return value (that isn't used, from what I see)?

Reply:
the return value indicates if the tap was consumed; this is also used that way

Reply:
Ah, nevermind, I didn't see it was there before, but in another method 🙈